src/main/java/org/isf/opd/rest/OpdController.java

```diff
@@ -199,25 +308,40 @@ public ResponseEntity<List<OpdDTO>> getOpdByDates(
 	}
 	
 	/**
-	 * Get all {@link Opd}s associated to specified patient CODE.
-	 * @return a {@link List} of {@link Opd} or NO_CONTENT if there is no data found.
+	 * Get all {@link OpdWithOperatioRowDTO}s associated to specified patient CODE.
+	 * @return a {@link List} of {@link OpdWithOperatioRowDTO} or NO_CONTENT if there is no data found.
 	 * @throws OHServiceException
 	 */
 	@GetMapping(value = "/opds/patient/{pcode}", produces = MediaType.APPLICATION_JSON_VALUE)
-	public ResponseEntity<List<OpdDTO>> getOpdByPatient(@PathVariable("pcode") int pcode) throws OHServiceException {
+	public ResponseEntity<List<OpdWithOperatioRowDTO>> getOpdByPatient(@PathVariable("pcode") int pcode) throws OHServiceException {
 		LOGGER.info("Get opd associated to specified patient CODE: {}", pcode);
 
 		List<Opd> opds = opdManager.getOpdList(pcode);
-		List<OpdDTO> opdDTOs = new ArrayList<OpdDTO>();
+		List<OpdWithOperatioRowDTO> opdWithOperations = new ArrayList<OpdWithOperatioRowDTO>();
 		if (!opds.isEmpty()) {
-			opdDTOs = opds.stream().map(opd -> {
-				return mapper.map2DTO(opd);
+			opdWithOperations = opds.stream().map(opd -> {
+				OpdWithOperatioRowDTO opRows = new OpdWithOperatioRowDTO();
+				opRows.setOpdDTO(mapper.map2DTO(opd));
+				List<OperationRow> listOp = new ArrayList<OperationRow>();
+				try {
+					listOp = operationRowManager.getOperationRowByOpd(opd);
+				} catch (OHServiceException e) {
+					// TODO Auto-generated catch block
```

Comment:
This should be at a minimum a logger message to tract what happened.   Just dumping a stack trace is never a good idea.

Reply:
you can remove the comment //TODO then